File: src/main/java/com/example/backend/dbmigration/InitDB.java
package com.example.backend.dbmigration;

import com.example.backend.documents.TarotCard;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import io.mongock.api.annotations.*;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.mongodb.core.MongoTemplate;

import java.io.IOException;
import java.util.List;

@Configuration
@RequiredArgsConstructor
@ChangeUnit(order = "001", id = "cards_init", author = "leonid")
public class InitDB {

    private final MongoTemplate mongoTemplate;
    private final ObjectMapper objectMapper;
    private final DbMigrationProps dbMigrationProps;
    private final MigrationHelper migrationHelper;

    @BeforeExecution
    public void beforeExecution() throws IOException {
        Boolean recreateJsonMigrationInitFile = dbMigrationProps.getRecreateJsonMigrationInitFile();
        if (recreateJsonMigrationInitFile) {
            migrationHelper.recreateJsonDbInitFileFromSqlFlyWayFile();
        }
    }

    @RollbackBeforeExecution
    public void rollbackBeforeExecution() throws IOException {  }

    @Execution
    public void execution() throws IOException {
        List<TarotCard> tarotCards = objectMapper.readValue(
                dbMigrationProps.getInitDbJsonFile().getFile(),
                new TypeReference<List<TarotCard>>() {
                }
        );
        mongoTemplate.insertAll(tarotCards);
    }

    @RollbackExecution
    public void rollback() {
    }
}

File: src/main/kotlin/com/example/backend/dbmigration/InitDB.kt
package com.example.backend.dbmigration

import com.example.backend.documents.TarotCard
import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import io.mongock.api.annotations.*
import org.springframework.context.annotation.Configuration
import org.springframework.data.mongodb.core.ReactiveMongoTemplate
import org.springframework.stereotype.Component

@Component
//@ChangeUnit(order = "001", id = "cards_init", author = "leonid")
class InitDB(
    private val mongoTemplate: ReactiveMongoTemplate,
    private val objectMapper: ObjectMapper,
    private val dbMigrationProps: DbMigrationProps,
    private val migrationHelper: MigrationHelper,
) {

    @BeforeExecution
    fun beforeExecution() {
        val recreateJsonMigrationInitFile = dbMigrationProps.recreateJsonMigrationInitFile
        if (recreateJsonMigrationInitFile == true) {
            migrationHelper.recreateJsonDbInitFileFromSqlFlyWayFile()
        }
    }

    @RollbackBeforeExecution
    fun rollbackBeforeExecution() {
    }

    @Execution
    fun execution() {
        val tarotCards = objectMapper.readValue(
            dbMigrationProps.initDbJsonFile?.file,
            object : TypeReference<List<TarotCard?>?>() {
            }
        )
        mongoTemplate.insertAll(tarotCards!!).subscribe()
    }

    @RollbackExecution
    fun rollback() {
    }
}